plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java

```diff
@@ -114,89 +91,125 @@ public SshKeyPair genKeyPair(String host, String comment, String passPhrase) thr
         }
+        return new SshKeyPair(publicKey, privateKey);
     }
 
     @Override
     public void removeKeys(String host) throws SshKeyStoreException {
-        User user = getUser(getUserId());
         try {
-            Map<String, String> preferences = preferenceDao.getPreferences(user.getId());
-            preferences.remove(sshKeyAttributeName(host, PRIVATE));
-            preferences.remove(sshKeyAttributeName(host, PUBLIC));
-            preferenceDao.setPreferences(user.getId(), preferences);
-        } catch (NotFoundException | ServerException e) {
-            throw new SshKeyStoreException(String.format("Failed to remove keys for host '%s'.", host));
+            final Map<String, String> keys = getSshKeys();
+            keys.remove(sshKeyAttributeName(host, PRIVATE));
+            keys.remove(sshKeyAttributeName(host, PUBLIC));
+            updateSshKeys(keys);
+        } catch (ApiException | JsonParseException | IOException e) {
+            throw new SshKeyStoreException("Failed to remove keys for host '" + host + "'.");
         }
     }
 
     @Override
     public Set<String> getAll() throws SshKeyStoreException {
-        User user = getUser(getUserId());
-        Map<String, String> preferences;
+        Map<String, String> keys;
         try {
-            preferences = preferenceDao.getPreferences(user.getId());
-        } catch (ServerException e) {
-            throw new SshKeyStoreException("Failed to get all keys.");
+            keys = getSshKeys();
+        } catch (ApiException | IOException | JsonParseException e) {
+            throw new SshKeyStoreException("Failed to get keys", e);
         }
-
-
-        final Set<String> keys = new HashSet<String>();
+        if (keys == null) {
+            throw new SshKeyStoreException("Impossible get keys");
+        }
+        final Set<String> privateKeys = new HashSet<>();
         // Check only for private keys.
-        for (String str : preferences.keySet()) {
-            if (str.startsWith(PRIVATE_KEY_ATTRIBUTE_PREFIX)) {
-                keys.add(str.substring(PRIVATE_KEY_ATTRIBUTE_PREFIX.length()));
+        privateKeys.addAll(keys.keySet()
+                               .stream()
+                               .filter(str -> str.startsWith(PRIVATE_KEY_ATTRIBUTE_PREFIX))
+                               .map(str -> str.substring(PRIVATE_KEY_ATTRIBUTE_PREFIX.length()))
+                               .collect(Collectors.toList()));
+        return privateKeys;
+    }
+
+    private SshKey getKey(String host, int i) throws SshKeyStoreException {
+        Map<String, String> keys;
+        try {
+            keys = getSshKeys();
+        } catch (ApiException | IOException | JsonParseException e) {
+            throw new SshKeyStoreException("Failed to get key for host '" + host + "'.", e);
+        }
+        if (keys == null) {
+            throw new SshKeyStoreException("Impossible to get key for host '" + host + "'.");
```

Comment:
Why lines 185 and 188 have so similar error messages? Maybe messages should be equal?